lib/tirexs/query.ex

```diff
@@ -11,11 +11,15 @@ defmodule Tirexs.Query do
   end
 
   @doc false
-  def _query(options, query_opts\\[]) do
-    if is_list(options) do
-      query_opts = Enum.fetch!(options, 0)
-      options = extract_do(options, 1)
-    end
+  def _query(options, query_opts \\ [])
+
+  @doc false
+  def _query(options, _query_opts) when is_list(options) do
+    [query: extract(extract_do(options, 1)) ++ Enum.fetch!(options, 0)]
```

Comment:
This actually will not compile, because options is mutated, if vars it must be:
```
query_opts = Enum.fetch!(options, 0)
options = extract_do(options, 1)
[query: extract(options) ++ query_opts]
```

no issue though